snuba/datasets/replays_processor.py

```diff
@@ -27,146 +27,121 @@
 metrics = MetricsWrapper(environment.metrics, "replays.processor")
 from sentry_sdk import capture_exception
 
-EventDict = Mapping[Any, Any]
+ReplayEventDict = Mapping[Any, Any]
 RetentionDays = int
 
 
-class ReplaysProcessor(MessageProcessor):
-    PROMOTED_TAGS = {
-        "environment",
-        "sentry:release",
-        "sentry:user",
-        "sentry:dist",
-    }
+USER_FIELDS_PRECEDENCE = ("id", "username", "email", "ip_address")
+
 
+class ReplaysProcessor(MessageProcessor):
     def __extract_timestamp(self, field: int) -> datetime:
         timestamp = _ensure_valid_date(datetime.utcfromtimestamp(field))
         if timestamp is None:
             timestamp = datetime.utcnow()
         return timestamp
 
-    def _structure_and_validate_message(
-        self, message: Mapping[Any, Any]
-    ) -> Optional[Tuple[EventDict, RetentionDays]]:
-
-        event = message
-        data = event["data"]
-
-        try:
-            # We are purposely using a naive datetime here to work with the
-            # rest of the codebase. We can be confident that clients are only
-            # sending UTC dates.
-            retention_days = enforce_retention(
-                message["retention_days"], datetime.utcfromtimestamp(data["timestamp"])
-            )
-        except EventTooOld:
-            return None
-
-        return event, retention_days
-
-    def _process_base_event_values(
-        self, processed: MutableMapping[str, Any], event_dict: EventDict
-    ) -> MutableMapping[str, Any]:
-
-        processed["replay_id"] = str(uuid.UUID(event_dict["replay_id"]))
-        processed["project_id"] = event_dict["project_id"]
-
-        processed["sequence_id"] = event_dict["sequence_id"]
-        processed["trace_ids"] = event_dict["trace_ids"]
-
+    def _add_user_field_tag(
+        self, processed: Any, replay_event: ReplayEventDict
+    ) -> None:
+        if "user" not in replay_event:
+            return
```

Comment:
`user` and `user_hash` are not nullable in your schema.

Reply:
👍🏼